src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,310 @@
+      address(AaveV3Ethereum.COLLECTOR),
+      IERC20(AaveV3EthereumAssets.rETH_UNDERLYING).balanceOf(address(this))
+    );
+  }
+
+  function depositTokenIntoV2(address token, uint256 amount) external onlyAdminOrManager {
+    if (!allowedFromTokens[token]) revert InvalidToken();
+    IERC20(token).approve(address(AaveV2Ethereum.POOL), amount);
+    AaveV2Ethereum.POOL.deposit(token, amount, address(AaveV3Ethereum.COLLECTOR), 0);
+    emit DepositedIntoV2(token, amount);
+  }
+
+  function depositTokenIntoV3(address token, uint256 amount) external onlyAdminOrManager {
+    if (!allowedFromTokens[token]) revert InvalidToken();
+    IERC20(token).approve(address(AaveV3Ethereum.POOL), amount);
+    AaveV3Ethereum.POOL.deposit(token, amount, address(AaveV3Ethereum.COLLECTOR), 0);
+    emit DepositedIntoV3(token, amount);
+  }
+
+  function setAdmin(address _admin) external onlyAdmin {
+    address oldAdmin = admin;
+    admin = _admin;
+    emit AdminChanged(oldAdmin, admin);
+  }
+
+  function setManager(address _manager) external onlyAdmin {
+    if (_manager == address(0)) revert Invalid0xAddress();
+    address oldManager = manager;
+    manager = _manager;
+    emit ManagerChanged(oldManager, manager);
+  }
+
+  function removeManager() external onlyAdmin {
+    address oldManager = manager;
+    manager = address(0);
+    emit ManagerChanged(oldManager, manager);
+  }
+
+  function setAllowedFromToken(address token, address oracle, bool isEthBased, bool allowed) external onlyAdmin {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedFromTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setAllowedToToken(address token, address oracle, bool isEthBased, bool allowed) external onlyAdmin {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedToTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setMilkmanAddress(address _milkman) external onlyAdmin {
+    if (_milkman == address(0)) revert Invalid0xAddress();
+    milkman = _milkman;
+  }
+
+  /// @notice Transfer any tokens on this contract to Aave V3 Collector
+  /// @param tokens List of token addresses
+  function withdrawToCollector(address[] calldata tokens) external onlyAdminOrManager {
+    for (uint256 i = 0; i < tokens.length; ++i) {
+      IERC20(tokens[i]).safeTransfer(
+        address(AaveV3Ethereum.COLLECTOR),
+        IERC20(tokens[i]).balanceOf(address(this))
+      );
+    }
+  }
+
+  receive() external payable {}
+
+  /// @inheritdoc VersionedInitializable
+  function getRevision() internal pure override returns (uint256) {
+    return REVISION;
+  }
+
+  function _getPriceCheckerAndData(
```

Comment:
I would handle the slippage encoding in this function since that's gonna be the same for all price checkers